include/rocksdb/env.h

```diff
@@ -557,8 +557,9 @@ class RandomAccessFile {
   virtual Status InvalidateCache(size_t offset, size_t length) {
     return Status::NotSupported("InvalidateCache not supported.");
   }
-};
 
+  virtual intptr_t FileDescriptor() const { abort(); return -1; }
```

Comment:
`intptr_t` complicates the problem. Can we just do `int`? Or maybe `long`? Linux fd is int. I don't see how intptr_t make it more platform independent.

The naming convention is probably `file_descriptor()`.

I don't suggest we call abort(), because it is a public header, and it replies on the availability of abort() of their platform. I would not returning anything.

Reply:
@siying
1. In class `WinMmapReadableFile`, file_descriptor is HANDLE, which is 64 bits on 64 bits system, both `int` and `long` is 32 bits on 64 bits windows system. another solution is to don't use `CreateFile` on windows, but use posix `open` on windows and get file `HANDLE` from function `intptr_t _get_osfhandle(int fd)`.
2. Rename to `file_descriptor()`  is OK.
3. `file_descriptor()` should be `pure virtual` ideally, we let it `abort` is just to avoid adding dummy definition for some special implementation such as `LibradosRandomAccessFile` and `RandomAccessFile`... If it sillently returns `-1`, some bugs may be hidden and may produce undefined behavior in a later place.